### DISCUSSION TITLE
Could you put making CrudFilters generic on the road map?

### DISCUSSION THREAD

**Discussion by vinclou** [refinedev/refine#2591] [Ideas]
*** Proposed functionality ***

```tsx
const filters: CrudFilters<MyCustomFilters> = []
const filters: CrudFilters<Omit<MyCustomFilters, 'id'> & { id: string }> = []
```

###### Although as a side comment
It is not terribly difficult to just make ur own generic types for this one, but it would add some core flexibility to the library.

Where filter form values could be different from what is sent to the server. Stems from an issue where it was easier to 
compose an array of strings from a string, separated by a comma, than dealing with native form support for that through select.

**Comment by omeraplak** [comment]
Hey @vinclou ,
Great idea! But I'm not sure about DX. How can we get this type as generic? Do you have a suggestion? Thank you 🎯

**Reply by vinclou** [reply]
I will try to play around with it on the weekend, although can't say I'm a TS guru for it 🥲

`What I'm doing now is a wrapper based on your implementation, mostly just to avoid duplicate logic by making the utility function, this way I don't need TS intellisense to quickly type params`

```tsx
/**
* @example onSearch: params => getDefaultFilters(params)
*/
export function getDefaultFilters<T extends AnyRecord>(params: T) {
	const filters: CrudFilters = []
	for (const [key, value] of Object.entries(params)) {
		if (value) {
			filters.push({
				field: key,
				operator: 'eq',
				value: value
			})
		}
	}
	return filters
}
```

**Reply by omeraplak** [reply]
Thank you @vinclou 🎉 @aliemir, what do you think?
